Apps/Sandcastle/gallery/development/3D Models Articulations.html

```diff
@@ -0,0 +1,152 @@
+        padding: 4px;
+        border-radius: 4px;
+    }
+    #toolbar input {
+        vertical-align: middle;
+        padding-top: 2px;
+        padding-bottom: 2px;
+    }
+</style>
+<div id="cesiumContainer" class="fullSize"></div>
+<div id="loadingOverlay"><h1>Loading...</h1></div>
+<div id="toolbar">
+    <div>Articulation:
+        <select class="cesium-button" data-bind="options: articulations,
+                                                 optionsText: 'name',
+                                                 value: selectedArticulation"></select>
+    </div>
+    <table>
+        <tbody data-bind="foreach: stages">
+            <tr>
+                <td data-bind="text: name"></td>
+                <td>
+                    <input type="range" min="-3" max="3" step="0.01" data-bind="value: current,
+                                      valueUpdate: 'input',
+                                      attr: {
+                                          min: minimum,
+                                          max: maximum
+                                      }">
+                    <input type="text" size="2" data-bind="value: currentText">
+                </td>
+            </tr>
+        </tbody>
+    </table>
+</div>
+<script id="cesium_sandcastle_script">
+function startup(Cesium) {
+    'use strict';
+//Sandcastle_Begin
+// this can be changed to any glTF model
+var modelUrl = 'https://assets.agi.com/models/launchvehicle.glb';
+
+var viewModel = {
+    articulations: [],
+    stages: [],
+    selectedArticulation: undefined
+};
+
+Cesium.knockout.track(viewModel);
+
+Cesium.knockout.getObservable(viewModel, 'selectedArticulation').subscribe(function(newArticulation) {
+    viewModel.stages = newArticulation.stages;
+});
+
+var toolbar = document.getElementById('toolbar');
+Cesium.knockout.applyBindings(viewModel, toolbar);
+
+var viewer = new Cesium.Viewer('cesiumContainer');
+var scene = viewer.scene;
+
+var height = 220000.0;
+var origin = Cesium.Cartesian3.fromDegrees(-74.693, 28.243, height);
+var modelMatrix = Cesium.Transforms.headingPitchRollToFixedFrame(origin, new Cesium.HeadingPitchRoll());
+
+var model = scene.primitives.add(Cesium.Model.fromGltf({
+    url : modelUrl,
+    modelMatrix : modelMatrix,
+    minimumPixelSize : 128
+}));
+
+model.readyPromise.then(function(model) {
+    var camera = viewer.camera;
+
+    // Zoom to model
+    var controller = scene.screenSpaceCameraController;
+    var r = 2.0 * Math.max(model.boundingSphere.radius, camera.frustum.near);
+    controller.minimumZoomDistance = r * 0.2;
+
+    var center = Cesium.Matrix4.multiplyByPoint(model.modelMatrix, Cesium.Cartesian3.ZERO, new Cesium.Cartesian3());
+    var heading = Cesium.Math.toRadians(0.0);
+    var pitch = Cesium.Math.toRadians(-10.0);
+    camera.lookAt(center, new Cesium.HeadingPitchRange(heading, pitch, r * 0.8));
+
+    viewModel.articulations = Object.keys(model._runtime.articulationsByName).map(function(articulationName) {
```

Comment:
Should this Sandcastle still be using private properties?

Reply:
This is [how the node explorer works](https://github.com/AnalyticalGraphicsInc/cesium/blob/3f34fed5975ba35274cd1b4d90ff671977821084/Apps/Sandcastle/gallery/development/3D%20Models%20Node%20Explorer.html#L208) as well.  There is no non-leaky way to expose all of the needed information, short of constructing a large hierarchy of classes and structs that mirror the contents of GMDF.

Reply:
That said, I do want a good solution for this, but I've pushed that desire to #7831.